src/runtime/lpython_builtin.py

```diff
@@ -75,3 +75,18 @@ def bool(x: i32) -> bool:
         return False
     else:
         return True
+
+
+def len(s: str) -> i32:
+    """
+    Return the length of the string `s`.
+    """
+    if s == '':
+        return 0
+    count: i32
+    count = 0
+    end: i32
+    end = -1
+    while s[count:end] != '':
```

Comment:
This raises a codegen error when it hits this line 👇 

https://github.com/lcompilers/lpython/blob/19dd8a71ad4262587ff6e3c80482fdefdde86344/src/libasr/codegen/asr_to_llvm.cpp#L1022

Reply:
Right. `len` is a truly "builtin" function. It does not have an implementation using LPython. Implement it like this:
```python
def len(s: str) -> i32:
    """
    Return the length of the string `s`.
    """
    pass
```
The LLVM backend then has to provide its own implementation of `len` (which it already does).

Reply:
Other functions in this category are numpy's `size`, `shape`, etc. They all have to be implemented by the backend directly. Other functions like `chr` can (and should) be implemented in LPython.